components/user-mgt/org.wso2.carbon.user.mgt.workflow/src/main/java/org/wso2/carbon/user/mgt/workflow/userstore/UserStoreActionListener.java

```diff
@@ -110,8 +111,10 @@ public boolean doPreDeleteUser(String userName, UserStoreManager userStoreManage
             String domain = userStoreManager.getRealmConfiguration().getUserStoreProperty(UserCoreConstants.RealmConfig
                                                                                                   .PROPERTY_DOMAIN_NAME);
             int tenantId = userStoreManager.getTenantId() ;
+            String currentUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
             PrivilegedCarbonContext.startTenantFlow();
             PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId, true);
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(currentUser);
 
```

Comment:
What's the reason for getting the user name from current context and setting it to a new tenant context here? What would happen if a user is logged in on ST and this code segment gets executed?